dali/pipeline/operators/geometric/bb_flip.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/geometric/bb_flip.h"
+
+namespace dali {
+
+
+DALI_REGISTER_OPERATOR(BbFlip, BbFlip, CPU);
+
+
+DALI_SCHEMA(BbFlip)
+                .DocStr(R"code(Operator for horizontal flip (mirror) of bounding box.
+                               Input: Bounding box coordinates; in either [x, y, w, h]
+                               or [left, top, right, bottom] format. All coordinates are
+                               in the image coordinate system (i.e. 0.0-1.0))code")
+                .NumInput(1)
+                .NumOutput(1)
+                .AddArg("coordinates_type",
+                        R"code(True, for width-height representation.
+                               False for two-point (ltrb) representation.)code",
+                        DALI_BOOL)
+                .AddArg("flip_type",
```

Comment:
It would be good to add a default value for flip_type and coordinates_type and document this as well.

Reply:
This code has major problem - now it always flip, we cannot do the random flip.
This operator needs to accept flip_vertical value or list/tensor and flip_horizontal value or list/tensor. Otherwise we cannot do random flip.
Probably image flip operator is missing that too. Please update both if possible.

Reply:
Added randomness. I will add it also for image flip operator.

Regarding `Optional` vs `Mandatory` operators, in my opinion, since we already have such distinction, we should use it. As I saw in other operators, most of arguments are declared `Optional` and there doesn't seem to be evident scheme, when to use which. For now I will change these arguments to optional, as requested. But in future I will opt for either get rid of unnecessary distinction between `Optional` and `Mandatory` or set up some policy for it.